core/domain/rights_manager.py

```diff
@@ -184,18 +214,45 @@ def _save_activity_rights(
 # same as the activity id).
 # TODO(msl): get rid of inline imports by refactoring code
 def _update_exploration_summary(activity_rights):
+    """Updates the exploration summary field for an activity.
```

Comment:
That's not what this does, I think. It should be:

```
   Updates the summary model for the exploration corresponding to `activity_rights`.
```

Activities don't have "summary fields"; instead, the fields are stored in a separate "exploration summary" model. See core/domain/exp_domain and core/domain/exp_services for details.